pennylane/templates/layers/strongly_entangling.py

```diff
@@ -92,6 +92,17 @@ def __init__(self, weights, wires, ranges=None, imprimitive=None, do_queue=True)
                 self.ranges = [(l % (len(wires) - 1)) + 1 for l in range(self.n_layers)]
             else:
                 self.ranges = [0] * self.n_layers
+        else:
+            for r in ranges:
```

Comment:
I am on the fence about this check...it is always tricky to get the right balance of slowdown due to checks vs. help for the user to avoid wrong inputs. Happy to leave it in though. 

If you want, one could add a sentence to the docstrings where `ranges` is described, saying that it cannot be zero or M, since that would result in one wire for the 2-qubit gate.

Reply:
I tend to keep this check in. Otherwise, the runtime error message would not be very informative about what went wrong (or sometimes just stuck on a loop without error message if one is only trying to draw the circuit). Also, this check only happens on initialization and costs computation time of O(number of layers), I believe the cost would not be noticeable.

Reply:
@wongwsvincent something to keep in mind is that because of PennyLane's functional user-interface, initialization will occur every time that the QNode is executed 🙂 So something to keep in mind --- overheads can add up quickly.

(Long term, a strategy where we 'just in time compile' the function would help eliminate this overhead)

Reply:
@josh146 Ah ha, thanks a lot for the explanation! I feel like we should keep it in to return a more helpful error message. Should we leave it in for this round, and try to eliminate the overhead in the future?

Reply:
I benchmarked with and without checks and the difference is negligible (on average about 21.4s vs 21.3s, but fluctuations of repeating the benchmark on my machine vary that first decimal digit quite a bit). And @wongwsvincent is right, the error message without checks is horrendous.

Since not many users will use custom ranges, I propose we leave both checks in!

Reply:
Thanks so much for checking the computation time quickly @mariaschuld !